app/src/androidTest/java/ch/admin/bag/dp3t/FakeWorkerTest.java

```diff
@@ -0,0 +1,298 @@
+package ch.admin.bag.dp3t;
+
+import android.content.Context;
+import android.util.Log;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.work.*;
+import androidx.work.testing.SynchronousExecutor;
+import androidx.work.testing.TestDriver;
+import androidx.work.testing.WorkManagerTestInitHelper;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import org.dpppt.android.sdk.DP3T;
+import org.dpppt.android.sdk.internal.AppConfigManager;
+import org.dpppt.android.sdk.internal.logger.LogLevel;
+import org.dpppt.android.sdk.internal.logger.Logger;
+import org.dpppt.android.sdk.models.ApplicationInfo;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import ch.admin.bag.dp3t.networking.FakeWorker;
+import ch.admin.bag.dp3t.storage.SecureStorage;
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class FakeWorkerTest {
+
+	Context context;
+	MockWebServer server;
+	TestDriver testDriver;
+
+	@Before
+	public void setup() throws IOException {
+		context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+		Logger.init(context, LogLevel.DEBUG);
+
+		// Initialize WorkManager for instrumentation tests.
+		Configuration config = new Configuration.Builder()
+				// Set log level to Log.DEBUG to make it easier to debug
+				.setMinimumLoggingLevel(Log.DEBUG)
+				.setExecutor(new SynchronousExecutor())
+				.build();
+		WorkManagerTestInitHelper.initializeTestWorkManager(context, config);
+
+		testDriver = WorkManagerTestInitHelper.getTestDriver(context);
+
+		server = new MockWebServer();
+		server.start();
+
+		AppConfigManager appConfigManager = AppConfigManager.getInstance(context);
+		DP3T.init(context, new ApplicationInfo("test", server.url("/bucket/").toString(), server.url("/report/").toString()),
+				null);
+		appConfigManager.setTracingEnabled(false);
+		DP3T.clearData(context);
+		DP3T.init(context, new ApplicationInfo("test", server.url("/bucket/").toString(), server.url("/report/").toString()),
+				null);
+		appConfigManager.setTracingEnabled(true);
+
+		SecureStorage.getInstance(context).setTDummy(-1);
+	}
+
+	@Test
+	public void testInitialTDummy() throws Exception {
+		List<WorkInfo> initialWorkList = WorkManager.getInstance(context).getWorkInfosByTag(FakeWorker.WORK_TAG).get();
+		assertEquals(0, initialWorkList.size());
+
+		FakeWorker.safeStartFakeWorker(context);
+		// TDummy is initialized to a time in the future.
+		assert (SecureStorage.getInstance(context).getTDummy() > System.currentTimeMillis());
```

Comment:
we should always make sure to use assertTrue() instead of assert() because assert() checks can be disable by jvm flags